### ISSUE TITLE
matplotlib backend: switch to constrained_layout?

### ISSUE THREAD

**Issue by michaelaye** [pyviz/holoviews#3469]
It's probably a good idea to the much better layout engine, as activated by `constrained_layout=True` in a `plt.subplots()` call for example?

https://matplotlib.org/tutorials/intermediate/constrainedlayout_guide.html

This also removes the need for the `tight` parameter (indeed it actually clashes), but I haven't seen the need for it since using the new layout engine.

This issue is possibly related to, but not dependent on #1339 .

**Comment by philippjfr** [comment]
Thanks for the suggestion, I had followed the development of that matplotlib PR but never followed up on it when it was merged. This definitely seems like a good idea.

**Comment by jbednar** [comment]
It's still marked experimental on the [matplotlib website](https://matplotlib.org/tutorials/intermediate/constrainedlayout_guide.html), though...

**Comment by michaelaye** [comment]
yeah, at this point, i don't think it will ever come out of "experimental." i'm kinda giving up on MPL and i's quirks in only sporadically working dynamically in JLab. Focusing on Bokeh for now, even so that's of course also not without its quirks.
